File: BroadcastReceivers/LocalBroadcastManager/app/src/main/java/com/tomerpacific/broadcastreceiver/MyBroadcastReceiver.java
package com.tomerpacific.broadcastreceiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;
import android.widget.Toast;

public class MyBroadcastReceiver extends BroadcastReceiver {

    private final String TAG = BroadcastReceiver.class.getSimpleName();

    @Override
    public void onReceive(Context context, Intent intent) {
        String data = intent.getExtras().getString("data");
        Toast.makeText(context, "Broadcast Received with data " + data, Toast.LENGTH_LONG).show();
    }
}

File: BroadcastReceivers/LocalBroadcastManager/app/src/main/java/com/tomerpacific/broadcastreceiver/MyBroadcastReceiver.kt
package com.tomerpacific.broadcastreceiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.widget.Toast

class MyBroadcastReceiver : BroadcastReceiver() {
    private val TAG = BroadcastReceiver::class.java.simpleName
    override fun onReceive(context: Context, intent: Intent) {
        val data = intent.extras?.getString("data")
        Toast.makeText(context, "Broadcast Received with data $data", Toast.LENGTH_LONG).show()
    }
}